src/algebra/group_power/lemmas.lean

```diff
@@ -459,6 +460,54 @@ by rw [← gmultiples_hom_symm_apply, ← gmultiples_hom_apply, equiv.apply_symm
 
 /-! `add_monoid_hom.ext_int` is defined in `data.int.cast` -/
 
+variables (M G A)
+
+/-- If `M` is commutative, `powers_hom` is a multiplicative equivalence. -/
+def powers_mul_hom [comm_monoid M] : M ≃* (multiplicative ℕ →* M) :=
```

Comment:
What I was originally looking for is the variant that doesn't require commutativity:
```suggestion
example [monoid M] : multiplicative ℕ →* M →* M := { to_fun := λ n x, x ^ n.to_add, ..sorry}
```
Does this exist elsewhere?

Reply:
Why do you think that it doesn't require commutativity? `(f n).map_mul` says `(x * y) ^ n.to_add = x ^ n.to_add * y ^ y.to_add`.

Reply:
Or did you mean `multiplicative ℕ →* M → M`?

Reply:
If yes, then one of the ways to get it is to define `monoid_hom.flip_pi : (M →* (α → N)) ≃ α → (M →* N)`.

Reply:
Add add `has_coe_t` from `mul_equiv` to `monoid_hom`.

Reply:
I think you're right, the variant I was originally looking for doesn't exist.

Reply:
It makes sense! We don't have `monoid_hom.flip_pi` yet.